test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -53,31 +55,81 @@ struct test_inclusive_scan_by_segment
         }
     }
 
-    template <typename Iterator1, typename Iterator2, typename Size>
+    template <typename TValue>
+    struct last_accum_state
+    {
+        int idx_begin = 0;
+        int idx_end = 0;
+        TValue summ = {};
+    };
+
+    template <typename TValue, typename Iterator>
+    int
+    accum_values(Iterator host_vals, int idx_begin, int idx_end, last_accum_state<TValue>& cash)
+    {
+        if (idx_begin == cash.idx_begin && idx_end == cash.idx_end + 1 && idx_end - 1 >= idx_begin)
+            cash.summ += host_vals[idx_end - 1];
+        else
+            cash.summ = ::std::accumulate(::std::addressof(host_vals[idx_begin]), ::std::addressof(host_vals[idx_begin]) + idx_end - idx_begin, 0);
+
+        cash.idx_begin = idx_begin;
+        cash.idx_end = idx_end;
+
+        return cash.summ;
+    }
+
+#ifdef DUMP_CHECK_RESULTS
+    template <typename Iterator, typename Size>
+    void display_param(const char* msg, Iterator it, Size n)
+    {
+        std::cout << msg;
+        for (int i = 0; i < n; ++i)
+        {
+            if (i > 0)
+                std::cout << ", ";
+            std::cout << it[i];
+        }
+        std::cout << std::endl;
+    }
+#endif // DUMP_CHECK_RESULTS
+
+    template <typename TValue, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
```

Comment:
I think we dont need `TValue` template parameter.

Reply:
Fixed